extension/src/views/gradleDaemons/services/GradleStatus.ts

```diff
@@ -0,0 +1,71 @@
+import { DaemonInfo } from "../models/DaemonInfo";
+import { DaemonStatus } from "../models/DaemonStatus";
+import { getGradleConfig } from "../../../util/config";
+import { GradleConfig } from "../../../proto/gradle_pb";
+import { GradleWrapper } from "./GradleWrapper";
+import { GradleLocalInstallation } from "./GradleLocalInstallation";
+import { GradleConnectionType } from "../models/GradleConnectionType";
+export class GradleStatus {
+    static async getConnectionType(gradleConfig: GradleConfig): Promise<GradleConnectionType> {
+        if (gradleConfig.getWrapperEnabled()) {
+            return GradleConnectionType.WRAPPER;
+        } else {
+            if (gradleConfig.getVersion()) {
+                return GradleConnectionType.SPECIFICVERSION;
+            } else if (gradleConfig.getGradleHome()) {
+                return GradleConnectionType.LOCALINSTALLATION;
+            }
+            return GradleConnectionType.WRAPPER;
+        }
+    }
+
+    static async getDaemonsStatusOutput(gradleConfig: GradleConfig, projectRoot: string): Promise<string> {
+        const connectionType = await this.getConnectionType(gradleConfig);
+        switch (connectionType) {
+            case GradleConnectionType.WRAPPER:
+                if (await GradleWrapper.hasValidWrapper(projectRoot)) {
+                    const wrapper = new GradleWrapper(projectRoot);
+                    return wrapper.exec(["--status", "quiet"]);
+                }
+                return "";
+            case GradleConnectionType.LOCALINSTALLATION:
+                const localInstallation = new GradleLocalInstallation(gradleConfig.getGradleHome());
+                return localInstallation.exec(["--status", "quiet"]);
+            case GradleConnectionType.SPECIFICVERSION:
+                return "";
+            default:
+                throw new Error("Unknown connection type");
+        }
+    }
+
+    static async getDaemonsStatusList(projectRoot: string): Promise<DaemonInfo[]> {
+        const gradleConfig = getGradleConfig();
+        const output = await this.getDaemonsStatusOutput(gradleConfig, projectRoot);
+
+        return this.parseDaemonInfo(output);
+    }
+
+    private static parseDaemonInfo(output: string): DaemonInfo[] {
+        if (!output) return [];
+
+        const lines = output.split("\n");
```

Comment:
On Windows, the line separator is `\r\n`, please take it into consideration as well.